src/test/java/games/strategy/engine/framework/ArgParserTest.java

```diff
@@ -0,0 +1,85 @@
+package games.strategy.engine.framework;
+
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+
+import org.junit.Test;
+
+public class ArgParserTest {
+
+
+  @Test
+  public void argsTurnIntoSystemProps() {
+    assertThat("check precondition, system property for our test key should not be set yet.",
+        System.getProperty(TestData.propKey), nullValue());
+
+    boolean result = ArgParser.handleCommandLineArgs(
+        TestData.sampleArgInput, TestData.samplePropertyNameSet);
+
+    assertThat("prop key was supplied as an available value, "
+            + " which was passed as a test value - everything should "
+            + "have parsed well, expect true result",
+        result, is(true));
+    assertThat("system property should now be set to our test value",
+        System.getProperty(TestData.propKey), is(TestData.propValue));
```

Comment:
Appreciate the feedback. Please allow me to offer some reasons of why  I did not clean up the system props:
- should not matter to one extent. Other tests should assert a precondition state, and if they do not check that props are not yet set, then they also have a bug too.
- that is perhaps not terribly convincing, more convincing is that we are not setting very many props. The one of consequence is likely set in other places too.

I do think the overkill route is roughly in the right direction. Instead of accessing and getting system properties by `String`, I've tried to update that to be by enum value instead. In subsequent iterations I would like System props to not be used if at all, instead making configuration objects available via `ClientContext`.

In the meantime, since I don't feel that strongly about this. I added in 6166482
 an @After block to this test case to clean up the one potentially important prop that this test leaks. Which regardless of my point of view here, I hope addresses the concern raised.